staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -657,6 +657,11 @@ func (c *Cacher) GetList(ctx context.Context, key string, opts storage.ListOptio
 		return c.storage.GetList(ctx, key, opts, listObj)
 	}
 
+	match := opts.ResourceVersionMatch
+	if match != metav1.ResourceVersionMatchNotOlderThan && match != "" {
```

Comment:
should `shouldDelegateList` change to this logic as well, so we delegate non-empty ResourceVersionMatch values the cacher does not understand?

Reply:
oops - sorry. Not sure, why I missed that.

Yes - that's a good point. Fixed here (and in the place that should be kept in sync).

Reply:
Thanks, update looks good. I think you could even drop this code and rely on `shouldDelegateList` only letting us reach this point for match values we support, since this error should be unreachable now. Up to you if you want to simplify that rather than keep two places in sync

Reply:
I have further test improvements in my local branch to clean them up, so I will clean this up in the following PR.
Thanks for pointing at it.